src/rai/rai/tools/ros/manipulation.py

```diff
@@ -0,0 +1,191 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import Literal, Type
+
+import numpy as np
+import rclpy
+import rclpy.callback_groups
+import rclpy.executors
+import rclpy.qos
+import rclpy.subscription
+import rclpy.task
+from geometry_msgs.msg import Point, Pose, PoseStamped, Quaternion
+from langchain_core.tools import BaseTool
+from pydantic import BaseModel, Field
+from rai_open_set_vision.tools import GetGrabbingPointTool
+from rclpy.client import Client
+from rclpy.node import Node
+from tf2_geometry_msgs import do_transform_pose
+
+from rai.tools.utils import TF2TransformFetcher
+from rai_interfaces.srv import ManipulatorMoveTo
+
+
+class MoveToPointToolInput(BaseModel):
+    x: float = Field(description="The x coordinate of the point to move to")
+    y: float = Field(description="The y coordinate of the point to move to")
+    z: float = Field(description="The z coordinate of the point to move to")
+    task: Literal["grab", "drop"] = Field(
+        description="Specify the intended action: use 'grab' to pick up an object, or 'drop' to release it. "
+        "This determines the gripper's behavior during the movement."
+    )
+
+
+class MoveToPointTool(BaseTool):
+    name: str = "move_to_point"
+    description: str = (
+        "Guide the robot's end effector to a specific point within the manipulator's operational space. "
+        "This tool ensures precise movement to the desired location. "
+        "While it confirms successful positioning, please note that it doesn't provide feedback on the "
+        "success of grabbing or releasing objects. Use additional sensors or tools for that information."
+    )
+
+    node: Node
+    client: Client
+
+    manipulator_frame: str = Field(
+        default="panda_link0", description="Manipulator frame"
+    )
+    min_z: float = Field(default=0.135, description="Minimum z coordinate [m]")
+    calibration_x: float = Field(default=0.071, description="Calibration x [m]")
+    calibration_y: float = Field(default=-0.025, description="Calibration y [m]")
+    calibration_z: float = Field(default=0.148, description="Calibration z [m]")
+    additional_height: float = Field(
+        default=0.05, description="Additional height for the place task [m]"
+    )
+
+    # constant quaternion
+    quaternion: Quaternion = Field(
+        default=Quaternion(x=0.9238795325112867, y=-0.3826834323650898, z=0.0, w=0.0),
+        description="Constant quaternion",
+    )
+
+    args_schema: Type[MoveToPointToolInput] = MoveToPointToolInput
+
+    def __init__(self, node: Node):
+        super().__init__(
+            node=node,
+            client=node.create_client(
```

Comment:
MoveToPoint should be an action in my opinion

Reply:
Noted